### PR TITLE
Particle Rounds

### PR DESCRIPTION
**Pull request by IProduceWidgets**
## About the PR

This PR adds Particle Bolts as projectiles. They sound not exactly, but very similarly, to how you expect them to.

(gif too big, sorry. Click for cool laser action.)
https://i.imgur.com/mn6mQtF.gif

Currently WIP as there is some additional functionality needed for what I want them to do.

They are 'plasma rifle' style projectiles meant to be fired out of a new class of Particle Weapons.
The plan is to make Particle Weapons explode if you try to shoot conventional projectiles out of them, and make conventional weapons explode if you try to shoot particle projectiles out of them. Unfortunately the functionality for that was recently reverted, but is also needed for flare guns, so I plan to make that work.

This PR also adds some functionality to Frezon which was missing, making it possible to digest it. It also makes Frezon make you... quite cold if you do. Also a Frezon reagent recolor because the current one is very bleh and dark, where as the frezon gas most often thought of is quite vibrant. 

proof all new things worky:
image

here are the colors of the particle bolts (the dark blue is the old color of frezon, the cyan is the new color of frezon):
image

## Why / Balance

The have the same nominal damage totals as their conventional counterparts. Where they differ is in how that damage is done.

Their goal is to be hell on hardsuits. While the majority of their damage comes from heat, their reagent effects are not impacted by armor, and in the case of frezon actively punish hardsuit wearers.

The base Particle Bolt does a mix of Heat and Slash damage.
The three ignoble gas particle bolts do Heat damage, but also poison their targets with the relevant reagent.

For Plasma Bolts this means Heat Damage plus ~6 poison / u
For Tritium Bolts this means Heat Damage plus ~6 radiation / u
For Frezon Bolts this means Cold Damage plus you actually get cold. This is kryptonite for people in hardsuits.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

### CODE REVIEW

**Comment by CrigCrag** [comment]
the frezon changes would probably be best put in another PR as they seem to be unrelated

**Comment by IProduceWidgets** [comment]
> the frezon changes would probably be best put in another PR as they seem to be unrelated

The projectile makes no sense without the changes, its very related.

**Comment by metalgearsloth** [comment]
I don't see a good argument why every single bullet is being copy-pasted for this?

File: Resources/Textures/Objects/Weapons/Guns/Projectiles/particleproj.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	  "name": "fill1"
+	},
+	{
+	  "name": "fill2"
+	},
+	{
+	  "name": "front"
+	}
+  ]
+}
```
Comment (metalgearsloth) [inline Resources/Textures/Objects/Weapons/Guns/Projectiles/particleproj.rsi/meta.json:23]: This copyright is wrong, you did not make particlebolt it is from tg.

Comment (IProduceWidgets) [reply Resources/Textures/Objects/Weapons/Guns/Projectiles/particleproj.rsi/meta.json:23]: Ahh, good catch, originally that rsi only had the fill layers and I ended up changing that. Will adjust.

**Comment by IProduceWidgets** [comment]
> I don't see a good argument why every single bullet is being copy-pasted for this?

the reason is so the calibers match the conventional projectile calibers so they can be (mistakenly) fired from the wrong weapons to make them explode once that functionality is back.

**Comment by IProduceWidgets** [comment]
Also, I'm willing to compromise on the name of them, but It seemed like all the good names are already taken between lasers and ions. Particle was the best I could come up with given that.

**Comment by metalgearsloth** [comment]
> Also, I'm willing to compromise on the name of them, but It seemed like all the good names are already taken between lasers and ions. Particle was the best I could come up with given that.

As is this just significantly increases yaml debt for minimal gameplay, you should start with a couple prototypes and give it a dedicated gun instead of doubling the existing ones.

**Comment by IProduceWidgets** [comment]
Hmm. Okay, I can drop all but the .35 pistol ones pretty easy.

I assume its okay / preferred if I don't try to mess with the gun system stuff in this PR for that wrong-ammo functionality and open another instead?

**Comment by metalgearsloth** [comment]
> Hmm. Okay, I can drop all but the .35 pistol ones pretty easy.
> 
> I assume its okay / preferred if I don't try to mess with the gun system stuff in this PR for that wrong-ammo functionality and open another instead?

Up to you I don't mind, my main worry is it's a lot of unused prototypes when I think this is better off being its own gun + sprite + bullets.

**Comment by IProduceWidgets** [comment]
Yeah. I do intend to add sprites for unique guns that shoot them, and I've been debating a silver/grey casing to make them visually different. I wanted to float the idea first though before I put in all the effort for the extra bits. 

Another idea I thought about, but haven't totally wrapped my head around would be some sort of GasTankAmmoSystem that would mean no cartridges etc., but it also means the ammo is a lot less restrict-able which I dunno if is good.

**Comment by metalgearsloth** [comment]
Making a gastank thing to shoot atmos gas sounds better.

**Comment by Emisse** [comment]
Closed due to feature freeze May 10th-June 14th. Comment to have it reopen after this.

**Pull request closed without merging**